zebrad/src/components/inbound/downloads.rs

```diff
@@ -145,17 +154,18 @@ where
     ZS: Service<zs::Request, Response = zs::Response, Error = BoxError> + Send + Clone + 'static,
     ZS::Future: Send,
 {
-    /// Initialize a new download stream with the provided `network` and
-    /// `verifier` services.
+    /// Initialize a new download stream with the provided `network`, `verifier`, and `state` services.
+    /// The `latest_chain_tip` must be linked to the provided `state` service.
```

Comment:
This seems like the sort of thing that would benefit from being enforced by the compiler.

Not something for this PR, and I'm also not sure how to do this. We'd need some type that contains the state service and the chain tip, or have a trait that extends the state service trait with a chain tip getter :thinking:

Reply:
We could add a request to the state service that returns a `ChainTipChange` and `LatestChainTip`, but that could make initialization awkward.

So a struct is probably the way to go, as long as it is generic over the exact state service type. (So we can wrap the state service in tower layers.)

Reply:
See #3198.